src/Fixer/Phpdoc/PhpdocTypesOrderFixer.php

```diff
@@ -0,0 +1,213 @@
+                    '<?php
+/**
+ * @param null|string|int|\Foo $bar
+ */',
+                    [
+                        'sort_algorithm' => 'alpha',
+                        'null_position' => 'default',
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('sort_algorithm', 'The sorting algorithm to apply.'))
+                ->setAllowedValues(['alpha', 'none'])
+                ->setDefault('none')
+                ->getOption(),
+            (new FixerOptionBuilder('null_position', 'Forces the position of `null`.'))
+                ->setAllowedValues(['always_first', 'always_last', 'default'])
+                ->setDefault('always_first')
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $annotations = $doc->getAnnotationsOfType(Annotation::getTagsWithTypes());
+
+            if (empty($annotations)) {
+                continue;
+            }
+
+            foreach ($annotations as $annotation) {
+                $types = $annotation->getTypes();
+
+                // fix main types
+                $annotation->setTypes($this->sortTypes($types));
+
+                // fix @method parameters types
+                $line = $doc->getLine($annotation->getStart());
+                $line->setContent(preg_replace_callback('/(@method\s+.+?\s+\w+\()(.*)\)/', function (array $matches) {
+                    $sorted = preg_replace_callback('/((?:^|,)\s*)([^\s]+)/', function (array $matches) {
+                        return $matches[1].$this->sortJoinedTypes($matches[2]);
+                    }, $matches[2]);
+
+                    return $matches[1].$sorted.')';
+                }, $line->getContent()));
+            }
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $doc->getContent()]);
+        }
+    }
+
+    /**
+     * @param string[] $types
+     *
+     * @return string[]
+     */
+    private function sortTypes(array $types)
+    {
+        foreach ($types as $index => $type) {
+            $types[$index] = preg_replace_callback('/^([^<]+)<(?:(.+?)(,\s*))?(.*)>$/', function (array $matches) {
+                return $matches[1].'<'.$this->sortJoinedTypes($matches[2]).$matches[3].$this->sortJoinedTypes($matches[4]).'>';
+            }, $type);
+        }
+
+        if ('alpha' === $this->configuration['sort_algorithm']) {
+            if (PHP_VERSION_ID < 70000) {
```

Comment:
Because of the way [sorting order of equal elements has changed with PHP 7](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.other.sort-order):
> The internal sorting algorithm has been improved, what may result in different sort order of elements, which compare as equal, than before.

On PHP 7, equal elements will remain in the same order while in PHP 5 the order is unpredictable. I'll add a comment with the link above.